src/vision/shiny_future/alan_switches_runtimes.md

```diff
@@ -0,0 +1,150 @@
+
+
+
+## Learning more about Humboldt
+
+Alan sees that some of the networking code that is being used in their application is creating network connections using humboldt APIs:
+
+```rust
+use humboldt::network;
+```
+
+He asks Barbara, "Why don't we use the `std::async_io` APIs for that?" She explains that Humboldt makes use of some custom kernel extensions that, naturally enough, aren't part of the std library. "TCP is for rubes," she says, "we are using TTCP -- Turbo TCP." Her mind wanders briefly to [Turbo Pascal] and she has a brief moment of yearning for the days when computers had a "Turbo" button that changed them from 8 MHz to 12 MHz. She snaps back into the present day. "Anyway, the `std::async_io` APIs just call into humboldt's APIs via various traits. But we can code directly against `humboldt` when we want to access the extra capabilities it offers."
+
+[Turbo Pascal]: https://en.wikipedia.org/wiki/Turbo_Pascal
+
+## Integrating into other event loops
+
+Later on, Alan is working on a visualizer front-end that integrates with [DistriData] to give more details about their workloads. To do it, he needs to integrate with Cocoa APIs and he wants to run certain tasks on [Grand Central Dispatch]. He approaches Barbara and asks, "If everything is running on `humboldt`, is there a way for me to run some things on another event loop? How does that work?"
+
+[Grand Central Dispatch]: https://en.wikipedia.org/wiki/Grand_Central_Dispatch
+
+Barbara explains, "That's easy. You just have to use the `gcd` wrapper crate -- you can find it on `crates.io`. It implements the runtime traits for `gcd` and it has a `spawn` method. Once you spawn your task onto `gcd`, everything you run within `gcd` will be running in that context."
+
+Alan says, "And so, if I want to get things running on `humboldt` again, I spawn a task back on `humboldt`?"
+
+"Exactly," says Barbara. "Humboldt has a global event loop, so you can do that by just doing `humboldt::spawn`. You can also just use the `humboldt::io` APIs directly. They will always use the Humboldt I/O threads, rather than using the current runtime."
+
+Alan winds up with some code that looks like this:
+
+```rust
+async fn do_something_on_humboldt() {
+    gcd::spawn(async move {
+        let foo = do_something_on_gcd();
+
+        let bar = humboldt::spawn(async move {
+            do_a_little_bit_of_stuff_on_humboldt();
+        });
+
+        combine(foo.await, bar.await);
+    });
+}
```

Comment:
Shouldn't `humboldt::spawn` be out of `gcd::spawn`?

Reply:
I wonder if a more realistic example would be to have multiple thread-pools, with e.g. different priorities or something. I'm going to leave it how it is, regardless.